hotsos/client.py

```diff
@@ -410,4 +403,3 @@ def run(self):
                         self.summary.update(plugin, content.get(plugin))
         finally:
             log.name = 'hotsos.client'
-            self.teardown_global_env()
```

Comment:
So I just wanted to make it clear this call *mostly* removed the temp files, but for some reason these kernel plugin ones were always left behind:

$ find tmpzssaz5uq
tmpzssaz5uq
tmpzssaz5uq/kernelva2q73j_
tmpzssaz5uq/kernelva2q73j_/locks
tmpzssaz5uq/kernelva2q73j_/locks/cache_commands.lock
tmpzssaz5uq/kernelva2q73j_/locks/cache_all_global.lock
tmpzssaz5uq/kernelva2q73j_/caches
tmpzssaz5uq/kernelva2q73j_/caches/host_helpers_cli
tmpzssaz5uq/kernelva2q73j_/caches/host_helpers_cli/commands
tmpzssaz5uq/kernelva2q73j_/caches/host_helpers_cli/commands/hostname

And 'kernel' is the last one run based on PLUGIN_RUN_ORDER.

Which actually led me to change the order and see what happens... and it shows the last one run always leaves these same files behind.

So maybe there is some better way to fix this?

I actually found the real problem. Above in get() there is this call:
    CLIHelper().hostname()

Which is going to write to the hostname file in the temp dir. Since HotSOSConfig.plugin_tmp_dir is set it writes into the last plugins temp dir, which in all cases is the kernel one, re-creating some of the tree there. Clearing that variable solves the issue, at least for the directory. That was not obvious at all.

I'll look at the log file comment you left and see how to deal with that.